### PR TITLE
Add TypeScript plugin example

### LINKED ISSUE #3: TypeScript support
Ensure that PostCSS plugins can be written in TypeScript, via https://github.com/bazelbuild/rules_nodejs/blob/master/packages/typescript/docs/install.md.

Add an example plugin and test that the plugin works as expected.

### PR DESCRIPTION
**Pull request by rzhw**
This PR adds dependencies on TypeScript and required tooling, and adds a copy of the custom plugin example rewritten using TypeScript. It also adds a test that the TypeScript-ified plugin output is the same as the original.

Closes #3

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -13,3 +13,5 @@
 # limitations under the License.
 
 exports_files(["LICENSE"])
+
+exports_files(["tsconfig.json"], visibility = ["//visibility:public"])
```
Comment (nex3) [inline BUILD]: Nit: trailing newline (also a couple other files)

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      "types": [
+        "node"
+      ],
+      "target": "es2015",
+      "strict": true
+    },
+    "include": [
+      "examples/*.ts",
+    ]
+  }
```
Comment (nex3) [inline tsconfig.json]: Style nit: it looks like this is indented four spaces at the outer level, but two spaces internally

**Review by nex3 — APPROVED** [review]

File: BUILD (hunk 1 above)
Comment (rzhw) [reply BUILD]: Do you know if there's a way to enforce this? (Or is enforcing even the way to go)

File: tsconfig.json (hunk 2 above)
Comment (rzhw) [reply tsconfig.json]: Done

File: BUILD (hunk 1 above)
Comment (rzhw) [reply BUILD]: (Done)

Comment (nex3) [reply BUILD]: I don't know of one, unfortunately.

**Pull request merged**